src/OVAL/results/oval_resultTest.c

```diff
@@ -618,6 +621,13 @@ static oval_result_t eval_check_state(struct oval_test *test, void **args)
 	syschar_model = oval_result_system_get_syschar_model(SYSTEM);
 	ores_clear(&item_ores);
 
+	char *state_names = oval_test_get_state_names(test);
+	dI("To get result of '%s', collected items will be compared with these oval states: %s.",
+		oval_test_get_id(test), state_names);
+	dI("In test '%s' %s of the collected items must satisfy these states: %s.",
+		oval_test_get_id(test), oval_check_get_description(ste_check), state_names);
```

Comment:
I understand you. The fact is that in current code the situation that there will be empty string (`satisfy states: ''`) will never happen, because this function `eval_check_state` is called only when the test contains at least one state, so this message will be displayed only when there is some state defined.

However this is not obvious from the code at all. I will improve it and also more ensure that this situation will not happen.